src/test/java/com/ghilly/service/CountryServiceRestTest.java

```diff
@@ -0,0 +1,135 @@
+package com.ghilly.service;
+
+import com.ghilly.repository.CountryRepositoryRest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CountryServiceRestTest {
+    private static final String NAME = "USSR";
+    private static final int ID = 1;
+    private CountryRepositoryRest repository;
+    private CountryServiceRest service;
+
+
+    @BeforeEach
+    void init() {
+        repository = mock(CountryRepositoryRest.class);
+        service = new CountryServiceRest(repository);
+    }
+    @Test
+    void addAndReceiveCountry() {
+        when(repository.takeCountry(ID)).thenReturn(NAME);
+
+        service.add(NAME);
+
+        assertAll(
+                () -> assertEquals(NAME, repository.takeCountry(ID)),
+                () -> verify(repository).insert(NAME)
+        );
+    }
+
+    @Test
+    void getAllCountries() {
+        List<String> expected = new ArrayList<>(List.of("Afghanistan", "France", "China"));
```

Comment:
Why you use ArrayList, list.of return List

Reply:
+